Sources/App/Utilities/Health.swift

```diff
@@ -0,0 +1,200 @@
+// Health check API
+// see https://inadarei.github.io/rfc-healthcheck/ for more info
+
+import Vapor
+import MySQL
+import Redis
+
+// MARK: Protocol
+
+protocol HealthComponent {
+    // A system component identifier eg. mysql or system
+    static var componentName: String { get }
+
+    // A system component type eg. datastore, system, etc.
+    static var componentType: String { get }
+
+    // An optional descriptive name of the type of masurement made eg. connection or uptime
+    static var measurementName: String? { get }
+
+    // The function that does the heath check on a given service
+    static func isHealthy(on request: Request) -> EventLoopFuture<Health.Indicator>
```

Comment:
Perhaps we should rename this? the "is" prefix leads to think it returns a boolean